Trove.js

```diff
@@ -0,0 +1,337 @@
+			ZU.processDocuments(urls, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+
+}
+
+function scrape(doc, url) {
+	if (url.indexOf('/newspaper/article/') != -1) {
+		scrapeNewspaper(doc, url);
+	} else {
+		scrapeWork(doc, url);
+	}
+}
+
+function scrapeNewspaper(doc, url) {
+	// There's a BibTex citation embedded in the page, seems like the easiest way to get the details.
+
+	var bibtex = ZU.xpathText(doc, "//textarea[@id='bibtex-citation']");
+
+	var translator = Zotero.loadTranslator("import");
+	translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");
+	translator.setString(bibtex);
+
+	// Clean up the BibTex results and add some extra stuff.
+
+	translator.setHandler("itemDone", function (obj, item) {
+		var articleID = item.url.match(/nla\.news-article\d+/)[0];
+		item.itemType = 'newspaperArticle';
+		item.abstractNote = ZU.xpathText(doc, "//meta[@property='og:description']/@content");
+		delete item.type;
+		delete item.itemID;
+
+		// Attaching a PDF is a 2 stage process.
+		// First you need to tell the service to start generating the PDF.
+		// You get back a hash id to include in the PDF url.
+
+		var renditionURL = "/newspaper/rendition/"
+		var prepURL = renditionURL + articleID + '/level/3/prep'
+
+		ZU.HTTP.doGet(prepURL, function(hashID) {
+
+			// It takes some time to generate the pdfs after the prep call.
+			// You can ping the service to see if the PDF is ready,
+			// but when I tried this I was denied permission to the statusText of the response object 
+			// which is necessary because statusText == 'Locked' if the PDF isn't ready.
+			// So I took out the ping call completely and just put in a 5 second pause which seems to work ok.
+			// Note that the service itself uses a 5 second delay if the ping fails.
+
+			var timeout = Date.now() + 5000;
+			while (Date.now() < timeout){}
+			item.attachments.push({url: renditionURL + articleID + '.3.pdf?followup=' + hashID, title: 'Trove newspaper PDF', mimeType:'application/pdf'});
+
+			// Get the OCRd text and save in a note.
+			var textURL = renditionURL + articleID + '.txt';
+			ZU.HTTP.doGet(textURL, function(text) {
+				item.notes.push({note: text.trim()});
+				item.complete();
+			});
+
+		});
+	});
+	translator.translate();	
+}
+
+var troveTypes = {
+	"Book": "book",
+	"Article Article/Book chapter": "bookSection",
+	"Thesis": "thesis",
+	"Archived website": "webpage",
+	"Conference Proceedings": "book",
+	"Audio book": "book",
+	"Article": "journalArticle",
+	"Article Article/Journal or magazine article": "journalArticle",
+	"Article Article/Conference paper": "conferencePaper",
+	"Article Article/Report": "report",
+	"Photograph": "artwork",
+	"Poster, chart, other": "artwork",
+	"Art work": "artwork",
+	"Object": "artwork",
+	"Microform Photograph": "artwork",
+	"Microform Object": "artwork",
+	"Sound": "audioRecording",
+	"Video": "videoRecording",
+	"Printed music": "book",
+	"Map": "map",
+	"Unpublished": "manuscript",
+	"Published": "document"
```

Comment:
not a great choice, since it has very limited fields and is not cited well. We generally default to journalArticle or book.